consensus/safety-rules/src/t_safety_rules.rs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{ConsensusState, Error};
+use consensus_types::{
+    block::Block, block_data::BlockData, quorum_cert::QuorumCert, timeout::Timeout, vote::Vote,
+    vote_proposal::VoteProposal,
+};
+use libra_types::crypto_proxies::Signature;
+
+/// Interface for SafetyRules
+pub trait TSafetyRules<T> {
```

Comment:
Will add the send + sync. TSafetyRules -> Trait for Safety Rules, seems intuitive to me though happy to chat about alternatives.

Reply:
Actually do we need to specify Send + Sync here versus later? I think in the ideal case this ought not need to be Sync + Send but it is unclear of the implications for doing so.

Reply:
The struct is the actual SafetyRules implementation. I'm partial to TSafetyRules since as we often used such context in Hack to distinguish Traits. I cannot find a common convention in Rust although T* is not common at all it seems. Maybe ISafetyRules (I => upper case i) for Interface for SafetyRules?